tests/test_web_sendfile_functional.py

```diff
@@ -238,47 +243,62 @@ async def handler(request):
     app.router.add_get("/", handler)
     client = await aiohttp_client(app)
 
-    resp = await client.get("/")
+    resp = await client.get("/", headers={"Accept-Encoding": accept_encoding})
     assert resp.status == 200
-    body = await resp.read()
-    assert b"hello aiohttp\n" == body
-    assert resp.headers["Content-Type"] == "application/pdf"
-    assert resp.headers.get("Content-Encoding") == "gzip"
+    assert (
+        resp.headers["Content-Type"],
+        resp.headers.get("Content-Encoding"),
+        await resp.read(),
+    ) == ("application/pdf", expect_encoding, b"hello aiohttp\n")
     resp.close()
     await resp.release()
     await client.close()
 
 
-async def test_static_file_with_gziped_counter_part_enable_compression(
-    aiohttp_client: Any, sender: Any
+@pytest.mark.parametrize(
+    ("accept_encoding", "expect_encoding"),
+    [("gzip, deflate", "gzip"), ("gzip, deflate, br", "br")],
+)
+@pytest.mark.parametrize("forced_compression", [None, web.ContentCoding.gzip])
+async def test_static_file_with_encoding_and_enable_compression(
+    aiohttp_client: Any,
+    sender: Any,
+    accept_encoding: str,
+    expect_encoding: str,
+    forced_compression: Optional[web.ContentCoding],
```

Comment:
Let's avoid `Optional`
```suggestion
    forced_compression: "web.ContentCoding | None",
```

Reply:
I took the advice of @Dreamsorcerer not to worry about these, and also from the rest of the code base.  There are 561 uses of `Optional[]`, and only 14 `Union[]` with `None`.

If this is that important, perhaps you could do a PR with a search and replace and enable the lint rule.

Reply:
Okay, fair. Resolved.